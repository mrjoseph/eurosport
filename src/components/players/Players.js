import React from 'react';
import PlayerData from '../player-data/PlayerData';
import './players.css';

const Image = ({ picture, firstname, lastname }) =>
    <div><img src={picture} alt={`${firstname} ${lastname}`}/></div>;

const Gender = ({ sex }) => {
  if(sex === 'M') {
    return 'Mens'
  }
  return 'Womens';
};

const Players = (data) => {
  const { players } = data;
  return (
      <div className="container">
        {players.map((player) => {
          const {
            firstname,
            lastname,
            shortname,
            sex,
            country,
            picture,
            data,
          } = player;
          return <div className="player" key={player.shortname}>
            <Image
                picture={picture}
                firstname={firstname}
                lastname={lastname}
            />
            <div className="player-stats-container">
              <div className="player-title">
                <img className="flag" src={country.picture} alt={shortname}/>
                <h3>{`${firstname} ${lastname}`}</h3>
              </div>
              <span className="gender"> <Gender sex={sex} /></span>
              <PlayerData {...data} />
            </div>
          </div>
        })}
      </div>
  );
};

export default Players;
