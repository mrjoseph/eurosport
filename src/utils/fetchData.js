import data from '../headtohead.json';

export default { data };
