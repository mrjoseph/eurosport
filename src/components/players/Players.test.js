import React from 'react';
import renderer from 'react-test-renderer';
import Players from './Players';
import fetchData from '../../utils/fetchData';
describe('Players', () => {
  it('renders players component ', () => {
    const { data } = fetchData;
    const rendered = renderer.create(
        <Players {...data} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
