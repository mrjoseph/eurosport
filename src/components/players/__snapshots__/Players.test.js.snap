// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players renders players component  1`] = `
<div
  className="container"
>
  <div
    className="player"
  >
    <div>
      <img
        alt="Stan Wawrinka"
        src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
      />
    </div>
    <div
      className="player-stats-container"
    >
      <div
        className="player-title"
      >
        <img
          alt="S.WAW"
          className="flag"
          src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
        />
        <h3>
          Stan Wawrinka
        </h3>
      </div>
      <span
        className="gender"
      >
         
        Mens
      </span>
      <div
        className="player-data-container"
      >
        <div
          className="player-data-row"
        >
          <span>
            rank:
          </span>
           
          21
        </div>
        <div
          className="player-data-row"
        >
          <span>
            points:
          </span>
           
          1784
        </div>
        <div
          className="player-data-row"
        >
          <span>
            weight:
          </span>
           
          81 kg
        </div>
        <div
          className="player-data-row"
        >
          <span>
            height:
          </span>
           
          1.83 m
        </div>
        <div
          className="player-data-row"
        >
          <span>
            age:
          </span>
           
          33 years old
        </div>
        <div
          className="win-loss"
        >
          <div>
            Wins: 
            4
          </div>
          <div>
            Losses: 
            1
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="player"
  >
    <div>
      <img
        alt="Rafael Nadal"
        src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg"
      />
    </div>
    <div
      className="player-stats-container"
    >
      <div
        className="player-title"
      >
        <img
          alt="R.NAD"
          className="flag"
          src="https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png"
        />
        <h3>
          Rafael Nadal
        </h3>
      </div>
      <span
        className="gender"
      >
         
        Mens
      </span>
      <div
        className="player-data-container"
      >
        <div
          className="player-data-row"
        >
          <span>
            rank:
          </span>
           
          1
        </div>
        <div
          className="player-data-row"
        >
          <span>
            points:
          </span>
           
          1982
        </div>
        <div
          className="player-data-row"
        >
          <span>
            weight:
          </span>
           
          85 kg
        </div>
        <div
          className="player-data-row"
        >
          <span>
            height:
          </span>
           
          1.85 m
        </div>
        <div
          className="player-data-row"
        >
          <span>
            age:
          </span>
           
          33 years old
        </div>
        <div
          className="win-loss"
        >
          <div>
            Wins: 
            2
          </div>
          <div>
            Losses: 
            3
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
